### DISCUSSION TITLE
Is there any way to set the type of the related table to single?

### DISCUSSION THREAD

**Discussion by msickpaler** [supabase/supabase#14036] [Questions]
Is there a way to make the type of the table to which the relation is to be made single? If not, do you plan to implement it?

parent : child = 1 : N.
In the following query, the type of the child table is `Child | Child[] | null`. However, since `limit(1, { foreignTable: "child" })` specifies that it is one, I want to eliminate `Child[]`.
It would be useful to use something like `single("child")` or `single("child.grand_child")` to eliminate the array. Is there any way to do this?

```typescript
const { data, error } = await supabase
    .from("parent")
    .select(
      "*, child(*)"
    )
    .eq("id", 2)
    .limit(1, { foreignTable: "child" })
    .limit(1)
    .single();
```

# version
`@supabase/supabase-js: 2.20.0`

**Comment by silentworks** [comment]
This will be available in the next version of PostgREST. I don't think this has been released to any hosted projects as yet but will be available in the coming months. You can test it out in local Supabase using the Supabase CLI. https://supabase.com/blog/postgrest-11-prerelease#spreading-related-tables

**Reply by msickpaler** [reply]
Thanks for sharing. But I think this is for the case `parent : child = N : 1` or `parent : child = 1 : 1`.
My question is about the `parent : child = 1 : N` case. If I only need one in this DB configuration (e.g. to get the latest), I want to use `single()` and fix the type to one as well.
